azure/services/agentpools/agentpools.go

```diff
@@ -133,10 +134,16 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			},
 		}
 
+		if profile.EnableAutoScaling != nil && *profile.EnableAutoScaling {
+			// unset the nodepool count if autoscaling is enabled
+			normalizedProfile.Count = nil
+			existingProfile.Count = nil
+		}
+
 		// Diff and check if we require an update
 		diff := cmp.Diff(normalizedProfile, existingProfile)
 		if diff != "" {
-			log.V(2).Info(fmt.Sprintf("Update required (+new -old):\n%s", diff))
+			klog.V(2).Info(fmt.Sprintf("Update required (+new -old):\n%s", diff))
```

Comment:
why switch to klog here?

Reply:
will update in the description. Following the [managedclusters]( https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/managedclusters/managedclusters.go#L325)

`klogs` does a pretty print format 
```
I0120 21:01:07.657204       1 managedclusters.go:325] Update required (+new -old):
  &containerservice.ManagedCluster{
  	Response: {},
  	ManagedClusterProperties: &containerservice.ManagedClusterProperties{
  		ProvisioningState: nil,
  		PowerState:        nil,
  		MaxAgentPools:     nil,
- 		KubernetesVersion: &"1.22.4",
+ 		KubernetesVersion: &"1.21.2",
  		DNSPrefix:         nil,
  		FqdnSubdomain:     nil,
...
```

while in agentpools we were getting in one line with escape chars

Reply:
wonder if that's something we can fix in our logger instead... see #1806 

we've added traces to the logger and correlation IDs which are useful for debugging

Reply:
cc @devigned

Reply:
got it! I will remove the change from this PR and later take a look at how we can fix that in our logger